types/enzyme/enzyme-tests.tsx

```diff
@@ -859,6 +865,14 @@ function ReactWrapperTest() {
         reactWrapper = new ReactWrapper<MyComponentProps, MyComponentState>(<MyComponent stringProp="1" numberProp={1} />, undefined, { attachTo: document.createElement('div') });
         reactWrapper = new ReactWrapper<MyComponentProps, MyComponentState>(<MyComponent stringProp="1" numberProp={1} />, reactWrapper, { attachTo: document.createElement('div') });
     }
+
+    function test_component_type() {
+      const wrapper1 = shallow(<div><ComponentType stringProp={"S"} numberProp={1} /></div>);
+      wrapper1.find<MyComponentProps>(ComponentType).props().stringProp; // $ExpectType string
```

Comment:
Unfortunate that the type param needs to be supplied here, but from my testing this goes back to optional statics influencing overload selection on .find and friends for React's ComponentClass and FunctionComponent. If I swap toComponentType above to reference Enzyme's ComponentType instead, the inference lights up here correctly. That's not how real code will be written, unfortunately, and I don't see a way around it 😞

Reply:
Agree, it's unfortunate ; but at least it's better than nothing